marimo/_output/hypertext.py

```diff
@@ -51,7 +52,57 @@ class Html(MIME):
     _text: str
 
     def __init__(self, text: str) -> None:
+        """Initialize the HTML element.
+
+        Subclasses of HTML MUST call this method.
+        """
         self._text = text
+        # A list of the virtual file names referenced by this HTML element.
+        self._virtual_filenames: list[str] = []
+
+        from marimo._runtime.context import (
+            ContextNotInitializedError,
+            get_context,
+        )
+
+        try:
+            ctx = get_context()
+        except ContextNotInitializedError:
+            return
+
+        # Virtual File Refcounting
+        #
+        # HTML elements are responsible for maintaining the reference counts
+        # of virtual files: virtual files cannot be disposed when HTML elements
+        # reference them. For example, a user might cache HTML referencing a
+        # virtual file if they create it using functools.cache.
+        #
+        # flatten the text to make sure searching isn't broken by newlines
+        flat_text = flatten_string(self._text)
```

Comment:
for `mo.ui.altair_chart` - it may have a virtual file in the `data` which gets base64 encoded. could this make `virtual_filename in flat_text` fail?

Reply:
I've checked manually and the file is there in plain text, here's a snippet of what's there:

`{&quot;url&quot;: &quot;/@file/3726-108546-7FiyZh2K.csv&quot;, `

Reply:
Filename is alphanumeric plus dashes and period. We only check the basename (without the bytes), in this case we'd be checking the string `108546-7FiyZh2K.csv`. So it should be fine?